### ISSUE TITLE
e.ChatMessage.EmoteSet returns wrong emotes

### ISSUE THREAD

**Issue by JustMeDaFaq** [swiftyspiffy/TwitchLib#99]
Hi,
whem im using e.ChatMessage.EmoteSet and sending a emote twice:

Username: Kappa Kappa

Im not getting two Kappa
25:0-4,25:0-4 but 
25:0-4,6-10

Thats happening with every emote im sending twice in one message.

Much thanks for your help! Will be probably the last thing im going to need help for :)

**Comment by swiftyspiffy** [comment]
The way the emote set works is this. Given the following chat message:
![test](http://swfty.xyz/?id=8764La213)

You'll get the following emoteset:
```
28087:15-21/25:5-9,28-32
```

The string can be initially split by the `/` symbol. On the left side you have the WutFace data, and on the right side, you have the Kappa data for BOTH Kappa's.
So using: `28087:15-21`.
The first number on the left of the `:` symbol is the emote ID. So this is the ID of WutFace. If you paste that ID into this format: `https://static-cdn.jtvnw.net/emoticons/v1/{emote_id}/1.0` you can get the direct link as seen here: ![test](https://static-cdn.jtvnw.net/emoticons/v1/28087/1.0) .
The numbers on the right of the `:` symbol represent the start and end indices. So the WutFace emote starts on character index 15, and ends on character index 21.

The same can be applied to the right side of the `/`, the side representing the Kappas. The first number, `25` represents the ID of Kappa, so if we plug that ID into the emote link format above, we get ![test](https://static-cdn.jtvnw.net/emoticons/v1/25/1.0) . This time however, since there are two Kappas in my message, we get two ranges. the first set is 5-9 and the second set is 28-32. This means that the first Kappa appears on character index 5, and ends on character index 9, while the second Kappa in the message appears on character index 28 and ends on character index 32.

TL;DR:
The numbers represent the emote ID and the emote start and end indices. I am currently writing code to parse this for the library, and return a list of ChatEmote objects that should make your job much easier.

**Comment by swiftyspiffy** [comment]
Alright, looks like we're good. Access `EmoteSet` like you currently are but now instead of the string you pasted above, it will be an object. You'll want to then access the `Emotes` property, which is a list of `Emote` objects you can use. An Emote object has the following properties: Name, StartIndex, EndIndex, Id, and ImageUrl.

For reference, there's an example I added to the Example Application: https://github.com/swiftyspiffy/TwitchLib/blob/master/TwitchLibExample/TwitchLibExample/Form1.cs#L225

**Comment by JustMeDaFaq** [comment]
That works fawlessly for received messages! Thank you!
What about send messages?  Received event does not trigger when im sending through the twitchlib api. Only OnMessageSent - which does not have the new emote object but the old emoteset string, which is different to the old emoteset string on received messagges :)

**Comment by swiftyspiffy** [comment]
OnMessageSent is a custom event I wrote (not sourced from Twitch itself), and it required a bit of hacking to get it to work as expected. Because of this, the event doesn't get the same indexing data that OnMessageRecevied gets. I can look into it, but you might need to do it on your side, as the data simply isn't available through conventional means for sent messages.

**Comment by JustMeDaFaq** [comment]
Ah okay.
Having issues in bigger channels with the new emotes integration. Getting this message on many new chat messages;
Ausnahme ausgelöst: "System.FormatException" in mscorlib.dll

**Comment by swiftyspiffy** [comment]
How fresh is your code you are using? I accidentally had a bug in the code and I pushed a fix 2 hours ago. Can you try pulling the fresh code and give it a go?

**Comment by JustMeDaFaq** [comment]
The new version fixed it for me! Thanks!
